conf/opers.sh

```diff
@@ -0,0 +1,155 @@
+<class name="BanControl" commands="KILL GLINE KLINE ZLINE QLINE ELINE TLINE RLINE CHECK NICKLOCK NICKUNLOCK SHUN CLONES CBAN CLOSE" usermodes="*" chanmodes="*">
+<class name="OperChat" commands="WALLOPS GLOBOPS" usermodes="*" chanmodes="*" privs="users/mass-message">
+<class name="HostCloak" commands="SETHOST SETIDENT SETIDLE CHGNAME CHGHOST CHGIDENT" usermodes="*" chanmodes="*" privs="users/auspex">
+
+
+#-#-#-#-#-#-#-#-#-#-#-#-  OPERATOR COMPOSITION   -#-#-#-#-#-#-#-#-#-#-#
+#                                                                     #
+#   This is where you specify which types of operators you have on    #
+#   your server, as well as the commands they are allowed to use.     #
+#   This works alongside with the classes specified above.            #
+#                                                                     #
+
+<type
+    # name: Name of type. Used in actual server operator accounts below.
+    # Cannot contain spaces. If you would like a space, use
+    # the _ character instead and it will translate to a space on whois.
+    name="NetAdmin"
+
+    # classes: Classes (blocks above) that this type belongs to.
+    classes="SACommands OperChat BanControl HostCloak Shutdown ServerLink"
+
+    # vhost: Host opers of this type get when they log in (oper up). This is optional.
+    vhost="netadmin${INSP_NET_SUFFIX}"
+
+    # modes: User modes besides +o that are set on an oper of this type
+    # when they oper up. Used for snomasks and other things.
+    # Requires that m_opermodes.so be loaded.
+    modes="+s +cCqQ">
+
+#-#-#-#-#-#-#-#-#-#-#-  OPERATOR CONFIGURATION   -#-#-#-#-#-#-#-#-#-#-#
+#                                                                     #
+#   Opers are defined here. This is a very important section.         #
+#   Remember to only make operators out of trustworthy people.        #
+#                                                                     #
+EOF
+
+# Generate oper with Fingerprint
+if [ "${INSP_OPER_FINGERPRINT}" != "" ]; then
+cat <<EOF
+# Operator with a plaintext password and no comments, for easy copy & paste.
+<oper
+      # name: Oper login that is used to oper up (/oper name password).
+      # Remember: This is case sensitive.
+      name="${INSP_OPER_NAME:-oper}"
+
+      # ** ADVANCED ** This option is disabled by default.
+      # fingerprint: When using the m_sslinfo module, you may specify
+      # a key fingerprint here. This can be obtained by using the /sslinfo
+      # command while the module is loaded, and is also noticed on connect.
+      # This enhances security by verifying that the person opering up has
+      # a matching SSL client certificate, which is very difficult to
+      # forge (impossible unless preimage attacks on the hash exist).
+      # If m_sslinfo isn't loaded, this option will be ignored.
+      fingerprint="${INSP_OPER_FINGERPRINT}"
```

Comment:
@Adam- What do you think: Should we stay with direct inclusion of the environment variables or switch to inspircd variables and provide the environment variables as defines, like we did with network prefix and servername?

`${INSP_OPER_FINGERPRINT}` vs. `&operFingerprint;` with central defines.

I noticed it by reviewing my own work... Changing it would break the tests. Stay with it break the concept :confused: